src/funcCoreTools/funcHostTask.ts

```diff
@@ -62,14 +66,33 @@ export function stopFuncTaskIfRunning(workspaceFolder: vscode.WorkspaceFolder):
     }
 }
 
-export function getFuncPortFromTask(funcTask: vscode.Task | undefined): string {
-    let port: string = defaultFuncPort;
+export async function getFuncPortFromTask(funcTask: vscode.Task | undefined, projectPathOrTaskScope: string | vscode.WorkspaceFolder | vscode.TaskScope): Promise<string> {
```

Comment:
I think method name needs a change to make it more general.

Reply:
`getFuncPortFromTaskOrProject`?

Personally, I think it's kind of a long-winded name but it is technically doing two things now.

I guess you could also write another function called `getFuncPortFromLocalSettings` that just does the new part that you added and call it in here as well. 🤷

Reply:
> I think function name needs a change to make it more general.

My preference when you review and don't like a name is that you make an alternative suggestion in your comment to "prove" there's a better name that still makes sense. As Nathan mentioned, even his better name is kinda long-winded and some functions are just hard to pick a name for.

> getFuncPortFromTaskOrProject

Sure